app/models/merge_request.rb

```diff
@@ -61,8 +61,29 @@ def total_visible_users_after_merge
     absorbing_organisation.users.visible.count + merging_organisations.sum { |org| org.users.visible.count }
   end
 
+  def total_stock_owners_after_merge
+    return total_stock_owners if status == STATUS[:request_merged] || status == STATUS[:processing]
+
+    stock_owners = absorbing_organisation.stock_owners.visible + merging_organisations.flat_map { |org| org.stock_owners.visible }
+    stock_owners.uniq.count
+  end
```

Comment:
If absorbing organisation was a stock owner of any of the merging organisations (or vice versa), after the merge happens I think that relationship would no longer be relevant, so we might want to filter out absorbing org and merging orgs from stock owners list before counting (?)

I think same with managing agents